core/arch/arm/plat-hikey/main.c

```diff
@@ -93,3 +108,166 @@ void console_flush(void)
 {
 	pl011_flush(console_base());
 }
+
+static vaddr_t peripherals_base(paddr_t pa)
+{
+	void *va;
+
+	if (cpu_mmu_enabled()) {
+		va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
```

Comment:
Why does `phys_to_virt` actually return a `void *` and not `vaddr_t`? That is a separate question and doesn't really belong to this patch, but in case we decide to make that change later, then it would make more sense to define `va` as `vaddr_t ...` instead and for now cast the return value from `phys_to_virt` to `vaddr_t` instead. Agree?